Flaky instrumentation tests break CI

The following PRs are currently failing because different instrumentation tests are failing:

- #223 (spotless-plugin-gradle upgrade). `tech.relaycorp.courier.ui.settings.SettingsActivityTest.setMaxStorageToMinimum` is failing with `java.lang.AssertionError: Timeout waiting for StorageSize(bytes=1000000000) to become StorageSize(bytes=100000000)`. https://console.firebase.google.com/project/relaynet-courier/testlab/histories/bh.c82b7d21521a430d/matrices/5049460515348117709/executions/bs.db8d21249adc0321/testcases/14/test-cases
- #225 (mockito-inline upgrade). `tech.relaycorp.courier.ui.sync.internet.InternetSyncActivityTest.showsDeliveringMessage` is erroring out with a `JobCancellationException`. https://console.firebase.google.com/project/relaynet-courier/testlab/histories/bh.c82b7d21521a430d/matrices/8844463936515791759/executions/bs.375bc3a2116b62a4/testcases/15/test-cases

Note that neither PR should affect the instrumentation tests at all.

Those tests are often failing on API 29 and, to a lesser extend, 24 and 26.

--- a/app/src/androidTest/java/tech/relaycorp/courier/test/BaseActivityTestRule.kt
+++ b/app/src/androidTest/java/tech/relaycorp/courier/test/BaseActivityTestRule.kt
@@ -3,8 +3,10 @@ package tech.relaycorp.courier.test
 import android.app.Activity
 import android.content.Intent
 import androidx.test.rule.ActivityTestRule
+import com.schibsted.spain.barista.rule.BaristaRule
 import com.schibsted.spain.barista.rule.cleardata.ClearFilesRule
 import com.schibsted.spain.barista.rule.cleardata.ClearPreferencesRule
+import com.schibsted.spain.barista.rule.flaky.FlakyTestRule
 import org.junit.rules.RuleChain
 import org.junit.rules.TestRule
 import org.junit.runner.Description
@@ -19,6 +21,9 @@ class BaseActivityTestRule<T : Activity>(
     private val clearPreferencesRule: ClearPreferencesRule = ClearPreferencesRule()
     private val clearDatabaseRule: ClearTestDatabaseRule = ClearTestDatabaseRule()
     private val clearFilesRule: ClearFilesRule = ClearFilesRule()
+    private val flakyTestRule: FlakyTestRule = FlakyTestRule().apply {
+        allowFlakyAttemptsByDefault(5)
+    }
     private val activityTestRule: ActivityTestRule<T> = ActivityTestRule(
         activityClass.java,
         true,
@@ -26,7 +31,9 @@ class BaseActivityTestRule<T : Activity>(
     )
 
     override fun apply(base: Statement, description: Description): Statement {
-        return RuleChain.outerRule(activityTestRule)
+        return RuleChain
+            .outerRule(flakyTestRule)
+            .around(activityTestRule)
             .around(clearPreferencesRule)
             .around(clearDatabaseRule)
             .around(clearFilesRule)

--- a/app/src/androidTest/java/tech/relaycorp/courier/test/BaseActivityTestRule.kt
+++ b/app/src/androidTest/java/tech/relaycorp/courier/test/BaseActivityTestRule.kt
@@ -3,7 +3,6 @@ package tech.relaycorp.courier.test
 import android.app.Activity
 import android.content.Intent
 import androidx.test.rule.ActivityTestRule
-import com.schibsted.spain.barista.rule.BaristaRule
 import com.schibsted.spain.barista.rule.cleardata.ClearFilesRule
 import com.schibsted.spain.barista.rule.cleardata.ClearPreferencesRule
 import com.schibsted.spain.barista.rule.flaky.FlakyTestRule

